### ISSUE TITLE
Wrong error message when for ,) inside anonymous function or in flatMap/map/flapTap etc 

### ISSUE THREAD

**Issue by lemastero** [scalastyle/scalastyle#356]
# Observed incorrect behavior:

```scala
class WrongError extends App {
  def foo: () => Map[String, Double] = () =>
    Map(
      "age of star HD 140283" -> 14.46,
      "age of Universe" -> 13.8,
    )
}
```

produce an incorrect error message:

`WrongError.scala: Expected token SEMI but got Token(ARROW,=>,119,=>)`

# Expected behavior:
provide proper error message sth like
`WrongError.scala: Expected token SEMI but got Token(RPAREN,),119,))`

In the following situation:

```scala
class WrongError extends App {
  Map(
    "age of star HD 140283" -> 14.46,
    "age of Universe" -> 13.8,
  )
}
```

gives `WrongError.scala: illegal start of simple expression: Token(RPAREN,),155,))` that correctly state problem but no line number.

# Severity:
Because there is no line number and the error message is misleading it is hard to find the issue

# Version
"org.scalastyle" %  "scalastyle_2.12" % "1.0.0"
Scala 2.12.10
sbt 1.3.5
used by: `"org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0"`

# Related:
* related: https://github.com/scalastyle/scalastyle/issues/191 https://github.com/scalastyle/scalastyle/issues/292
* line number issues: https://github.com/scalastyle/scalastyle/issues/187 https://github.com/scalastyle/scalastyle/issues/195 https://github.com/scalastyle/scalastyle/issues/319
* possible couse scala-ide/scalariform issue https://github.com/scala-ide/scalariform/issues/163
